File: server/src/game/tools/conf/output/go/PayConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type PayConfigElem struct {
	Id int
	Sort int
	ProductId string
	ProductName string
	ProductDesc string
	Visible bool
	Hot bool
	MoneyType int
	MoneyCount int
	GiftMoneyCount int
	Price float32
	Currency string
	
}

type PayConfig0 struct {
	Dic map[int]*PayConfigElem
	Arr []*PayConfigElem
}

var (
	PayConfig PayConfig0
)

func loadPayConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	PayConfig.Dic = make(map[int]*PayConfigElem)
	PayConfig.Arr = make([]*PayConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &PayConfig.Arr)
	for _,conf := range PayConfig.Arr{
		PayConfig.Dic[conf.Id] = conf
	}
	
}

File: client/src/game/conf/PayConfig.ts
class PayConfig {
    private static _dic:Dictionary<PayConfig> = new Dictionary<PayConfig>();
    private static _arr:Array<PayConfig> = new Array<PayConfig>();

	 public Id :number ;
     public Sort :number ;
     public ProductId :string ;
     public ProductName :string ;
     public ProductDesc :string ;
     public Visible :boolean ;
     public Hot :boolean ;
     public MoneyType :number ;
     public MoneyCount :number ;
     public GiftMoneyCount :number ;
     public Price :number ;
     public Currency :string ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<PayConfig>{
        return  PayConfig._dic;
    }
    public static get Arr():Array<PayConfig>{
        return PayConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:PayConfig = <PayConfig>json[i];
				PayConfig._arr.push(config);
				PayConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("PayConfig loader fail!");
        }
    }
}